File: backend/src/main/java/com/back/domain/member/member/dto/MemberJoinReqDto.java
package com.back.domain.member.member.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record MemberJoinReqDto(
    @NotBlank
    @Size(min = 2, max = 30)
    String email,
    @NotBlank
    @Size(min = 2, max = 20)
    String name,
    @NotBlank
    @Size(min = 2, max = 20)
    String password
) {}

File: backend/src/main/java/com/back/domain/member/member/dto/MemberJoinReqDto.kt
package com.back.domain.member.member.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class MemberJoinReqDto(
    @field:NotBlank
    @field:Size(min = 2, max = 30)
    val email: String,
    
    @field:NotBlank
    @field:Size(min = 2, max = 20)
    val name: String,
    
    @field:NotBlank
    @field:Size(min = 2, max = 20)
    val password: String
)
